consensus/polybft/signer/signature_test.go

```diff
@@ -26,6 +29,80 @@ func Test_VerifySignature(t *testing.T) {
 	assert.False(t, signature.Verify(blsKey.PublicKey(), invalidTestMsg))
 }
 
+func Test_VerifySignature_NegativeCases(t *testing.T) {
+	t.Parallel()
+
+	// Get a random integer between 1 and 1000
+	mRand.Seed(time.Now().UnixNano())
+	messageSize := mRand.Intn(1000) + 1
+
+	validTestMsg := testGenRandomBytes(t, messageSize)
+
+	blsKey, err := GenerateBlsKey()
+	require.NoError(t, err)
+
+	signature, err := blsKey.Sign(validTestMsg)
+	require.NoError(t, err)
+
+	require.True(t, signature.Verify(blsKey.PublicKey(), validTestMsg))
+
+	t.Run("Wrong public key", func(t *testing.T) {
+		t.Parallel()
+
+		for i := 0; i < 100; i++ {
+			x, randomG2, err := bn256.RandomG2(rand.Reader)
+			require.NoError(t, err)
+
+			publicKey := blsKey.PublicKey()
+			publicKey.g2.Add(publicKey.g2, randomG2) // change public key g2 point
+			require.False(t, signature.Verify(publicKey, validTestMsg))
+
+			publicKey = blsKey.PublicKey()
```

Comment:
I think you could just do `publicKey = &PublicKey{g2:x}` or even `publicKey.g2 = x`